### PR TITLE
cleanup: refactor Windows build scripts and configs

### PR DESCRIPTION
**Pull request by coryan**
Made the build scripts for Windows behave much like their Linux
counterparts. You can manually run a specific build using:

powershell -exec bypass ci\kokoro\windows\build.ps1 cmake

Likewise, on Kokoro the script determines the build configuration from
the KOKORO_JOB_NAME environment variable.

Also refactored as much as possible out of the .bat file, currently it
just loads the MSVC path configuration (which we must do from a .bat
file), and then jumps into a powershell script.

### CODE REVIEW

File: ci/kokoro/windows/cmake-presubmit.cfg [hunk 1]
```diff
@@ -13,5 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-build_file: "google-cloud-cpp-common/ci/kokoro/windows/bazel/build.bat"
-timeout_mins: 120
+env_vars {
+  key: "BUILD_CACHE"
+  value: "gs://cloud-cpp-kokoro-results/build-artifacts/common/vcpkg-installed-win-2019.zip"
```
Comment (tmatsuo) [inline ci/kokoro/windows/cmake-presubmit.cfg]: Nit: filename suggests it's using Windows 2019 image.

File: ci/kokoro/windows/cmake.cfg [hunk 2]
```diff
@@ -13,15 +13,6 @@
...
-
-action {
-  define_artifacts {
-    regex: "**/win_minidumps/*.dmp"
-  }
-}
-
 env_vars {
   key: "BUILD_CACHE"
   value: "gs://cloud-cpp-kokoro-results/build-artifacts/common/vcpkg-installed-win-2019.zip"
```
Comment (tmatsuo) [inline ci/kokoro/windows/cmake.cfg]: ditto

File: ci/kokoro/windows/cmake-presubmit.cfg (hunk 1 above)
Comment (coryan) [reply ci/kokoro/windows/cmake-presubmit.cfg]: Fixed, I think the new name makes it more obvious that it is related to the build more than anything else.

File: ci/kokoro/windows/cmake.cfg (hunk 2 above)
Comment (coryan) [reply ci/kokoro/windows/cmake.cfg]: Done.  Thanks.

**Comment by coryan** [comment]
PTAL.

File: ci/kokoro/windows/cmake-presubmit.cfg [hunk 3]
```diff
@@ -13,5 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-build_file: "google-cloud-cpp-common/ci/kokoro/windows/bazel/build.bat"
-timeout_mins: 120
+env_vars {
+  key: "BUILD_CACHE"
+  value: "gs://cloud-cpp-kokoro-results/build-artifacts/common/windows/cmake-vcpkg-installe.zip"
```
Comment (tmatsuo) [inline ci/kokoro/windows/cmake-presubmit.cfg]: nit: I guess it's a TYPO of `cmake-vcpkg-installed.zip`

File: ci/kokoro/windows/cmake.cfg [hunk 4]
```diff
@@ -13,16 +13,7 @@
...
-action {
-  define_artifacts {
-    regex: "**/win_minidumps/*.dmp"
-  }
-}
-
 env_vars {
   key: "BUILD_CACHE"
-  value: "gs://cloud-cpp-kokoro-results/build-artifacts/common/vcpkg-installed-win-2019.zip"
+  value: "gs://cloud-cpp-kokoro-results/build-artifacts/common/windows/cmake-vcpkg-installe.zip"
```
Comment (tmatsuo) [inline ci/kokoro/windows/cmake.cfg]: ditto

**Pull request merged**